.github/workflows/build-push-container-images.yml

```diff
@@ -350,77 +293,6 @@ jobs:
-    needs:
-      - build-jvm-images
-      - build-native-images
-      - build-ui-images
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        arch:
-          - amd64
-          - arm64
-    steps:
-      - name: Calculate Branch (workflow_run event)
-        if: github.event_name == 'workflow_run'
-        run: |
-          echo "REF=${{ github.event.workflow_run.head_commit.id }}" >> $GITHUB_ENV
-          echo "BRANCH=${{ github.event.workflow_run.head_branch }}" >> $GITHUB_ENV
-
-      - name: Calculate Branch (workflow_dispatch event)
-        if: github.event_name == 'workflow_dispatch'
-        run: |
-          echo "REF=${{ github.sha }}" >> $GITHUB_ENV
-          echo "BRANCH=${{ github.ref_name }}" >> $GITHUB_ENV
-
-      - name: Checkout from ${{ env.REF }}
-        uses: actions/checkout@v3
-        with:
-          ref: ${{ env.REF }}
-
-      - name: Setup Java
-        uses: actions/setup-java@v3
-        with:
-          java-version: 17
-          distribution: temurin
-          cache: maven
-
-      - name: Create env vars (main branch)
-        if: env.BRANCH == 'main'
-        working-directory: ui-super-heroes
-        run: |
-          echo "CONTAINER_TAG=${{ env.LATEST_IMAGE_TAG }}-${{ matrix.arch }}" >> $GITHUB_ENV &&
-          echo "APP_VERSION=$(./mvnw help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_ENV
-
-      - name: Create env vars (other branch)
-        if: env.BRANCH != 'main'
-        working-directory: ui-super-heroes
-        run: |
-          echo "CONTAINER_TAG=${{ env.LATEST_IMAGE_TAG }}-${{ env.BRANCH }}-${{ matrix.arch }}" >> $GITHUB_ENV &&
-          echo "APP_VERSION=$(./mvnw help:evaluate -Dexpression=project.version -q -DforceStdout)-${{ env.BRANCH }}" >> $GITHUB_ENV
-
-      - name: Get Saved UI Image (${{ matrix.arch }})
-        uses: ishworkh/docker-image-artifact-download@v1
-        with:
-          image: "${{ env.IMAGE_BASE_NAME }}/ui-super-heroes:${{ env.CONTAINER_TAG }}"
-
-      - name: Login to quay
-        uses: docker/login-action@v2
-        with:
-          registry: quay.io
-          username: ${{ secrets.QUAY_REPO_USERNAME }}
-          password: ${{ secrets.QUAY_REPO_TOKEN }}
-
-      - name: Tag UI image (${{ matrix.arch }})
-        working-directory: ui-super-heroes
-        run: docker tag ${{ env.IMAGE_BASE_NAME }}/ui-super-heroes:${{ env.CONTAINER_TAG }} ${{ env.IMAGE_BASE_NAME }}/ui-super-heroes:${{ env.APP_VERSION }}-${{ matrix.arch }}
-
-      - name: Push UI image (${{ matrix.arch }})
-        working-directory: ui-super-heroes
-        run: "docker push -a ${{ env.IMAGE_BASE_NAME }}/ui-super-heroes"
-
   create-app-multiarch-manifests:
```

Comment:
I did indeed, good spot. 

Although my error is more evidence that maybe one big matrix is the way to go. :)

Reply:
maybe now after the restructuring that does make more sense, although it is broken up because we want to build all the images of the matrix before pushing a single one, and then only create the multi-arch manifest only once all the images are pushed, so it may still need to remain as separate jobs.